src/main/java/com/gmail/yuramitryahin/service/AuthenticationServiceImpl.java

```diff
@@ -19,20 +16,13 @@ public AuthenticationServiceImpl(UserService userService,
         this.shoppingCartService = shoppingCartService;
     }
 
-    @Override
-    public User login(String email, String password) throws AuthenticationException {
-        Optional<User> optionalUser = userService.findByEmail(email);
-        if (optionalUser.isPresent()
-                && HashUtil.checkPassword(optionalUser.get().getPassword(),
-                password, optionalUser.get().getSalt())) {
-            return optionalUser.get();
-        }
-        throw new AuthenticationException("Incorrect login or password!");
-    }
-
     @Override
     public User register(String email, String password) {
-        User user = new User(email, password);
```

Comment:
You can have both the builder and your custom constructors. In this case, I would prefer to use a constructor.